pkg/build/controller/image_change_controller.go

```diff
@@ -99,68 +95,37 @@ func (c *ImageChangeController) HandleImageRepo(repo *imageapi.ImageStream) erro
 		}
 
 		if shouldBuild {
-			// The following update is meant to reduce the chance that the image change controller
-			// will kick off multiple builds on an image change in a HA setup, where multiple controllers
-			// of the same type may be looking at the same etcd data.
-			// If multiple controllers read the same build config (with same ResourceVersion) above and
-			// make a determination that a build needs to be kicked off, the update will only allow one of
-			// those controllers to continue to launch the build, while the rest will return an error and
-			// reset their queue. This won't eliminate the chance of multiple builds, since another controller
-			// can read the build after this update and launch its own build.
-			// TODO: Find a better mechanism to synchronize in a HA setup.
-			if err := c.BuildConfigUpdater.Update(originalConfig); err != nil {
-				// Cannot make an update to the original build config. Likely it has been changed by another process
-				glog.V(4).Infof("Cannot update BuildConfig %s/%s when preparing to update LastTriggeredImageID: %v", config.Namespace, config.Name, err)
-				return err
-			}
-
 			glog.V(4).Infof("Running build for BuildConfig %s/%s", config.Namespace, config.Name)
 			// instantiate new build
-			request := &buildapi.BuildRequest{ObjectMeta: kapi.ObjectMeta{Name: config.Name}}
-			if _, err := c.BuildConfigInstantiator.Instantiate(config.Namespace, request); err != nil {
-				return fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %v", config.Namespace, config.Name, err)
+			request := &buildapi.BuildRequest{
+				ObjectMeta: kapi.ObjectMeta{
+					Name: config.Name,
+				},
+				Image: triggeredImage,
 			}
-			// and update the config
-			if err := c.updateConfig(config); err != nil {
-				// This is not a retryable error. The worst case outcome of not updating the buildconfig
-				// is that we might rerun a build for the same "new" imageid change in the future,
-				// which is better than guaranteeing we run the build 2+ times by retrying it here.
-				return ImageChangeControllerFatalError{
-					Reason: fmt.Sprintf("error updating BuildConfig %s/%s with new LastTriggeredImageID", config.Namespace, config.Name),
-					Err:    err,
+			if _, err := c.BuildConfigInstantiator.Instantiate(config.Namespace, request); err != nil {
+				if kerrors.IsConflict(err) {
+					// This is not a retryable error. The worst case outcome of not updating the buildconfig
+					// is that we might rerun a build for the same "new" imageid change in the future,
+					// which is better than guaranteeing we run the build 2+ times by retrying it here.
+					return ImageChangeControllerFatalError{
+						Reason: fmt.Sprintf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", config.Namespace, config.Name, err),
+						Err:    err,
+					}
 				}
+
+				return fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %v", config.Namespace, config.Name, err)
```

Comment:
are things expecting `ImageChangeControllerFatalError` errors? Should this be a typed error?